app/main.py

```diff
@@ -1,2 +1,144 @@
+import math
+from collections.abc import Hashable
+from typing import Any, Dict, Generator, Iterable, Tuple
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.capacity = 8
+        self.load_factor = 2 / 3
+        self.threshold = math.floor(self.capacity * self.load_factor)
+        self.hash_table = [None] * self.capacity
+        self.length = 0
+
+    def __len__(self) -> int:
+        return self.length
+
+    def _resize(self) -> None:
+        self.capacity *= 2
+        self.threshold = math.floor(self.capacity * self.load_factor)
+        old_table = self.hash_table
+        self.hash_table = [None] * self.capacity
+        self.length = 0
+
+        for bucket in old_table:
+            if bucket is not None and bucket != "del":
+                key, key_hash, value = bucket
+                index = key_hash % self.capacity
+
+                for _ in range(self.capacity):
+                    target = self.hash_table[index]
+                    if target is None:
+                        self.hash_table[index] = bucket
+                        self.length += 1
+                        break
+                    index = (index + 1) % self.capacity
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        key_hash = hash(key)
+        index = key_hash % self.capacity
+
+        if self.length >= self.threshold:
+            target = self.hash_table[index]
+            if target is None or target == "del":
+                self._resize()
+                index = key_hash % self.capacity
+            elif target[0] != key:
+                self._resize()
+                index = key_hash % self.capacity
```

Comment:
The resizing logic is incorrect here. The `_resize()` method should be called before checking the target bucket, based on the load factor, not after checking if the target is `None` or `del`. This ensures that the dictionary resizes when the load factor threshold is reached, not based on the current bucket state.